### PR TITLE
Fix (or disable) remaining failg tests on Windows

### PR DESCRIPTION
**Pull request by bellenot**
Following the latest changes in ROOT, some test are now working (then remove the WILLFAIL flag), and mark the forgotten failing ones as WILLFAIL

### CODE REVIEW

File: root/dataframe/test_trainCache.C [hunk 1]
```diff
@@ -1,3 +1,5 @@
+R__LOAD_LIBRARY(libROOTDataFrame);
```
Comment (eguiraud) [inline root/dataframe/test_trainCache.C:1]: why is this required? or: why was it not required before?

File: root/dataframe/CMakeLists.txt [hunk 2]
```diff
@@ -24,19 +24,22 @@ ROOTTEST_ADD_TEST(test_trainCache
 # ROOT-9628
 ROOTTEST_ADD_TEST(test_chainZombieFile
                   COPY_TO_BUILDDIR input1.root input2.root
+                  ${MIGHT_FAIL}
```
Comment (eguiraud) [inline root/dataframe/CMakeLists.txt:27]: the variable name is misleading, should really be called `WILL_FAIL_ON_WIN` or similar

File: root/dataframe/test_trainCache.C (hunk 1 above)
Comment (bellenot) [reply root/dataframe/test_trainCache.C:1]: I don't know, but without that it fails on Windows...

File: root/dataframe/CMakeLists.txt (hunk 2 above)
Comment (bellenot) [reply root/dataframe/CMakeLists.txt:27]: It's already user all over the place, but I'll change them all...

Comment (eguiraud) [reply root/dataframe/CMakeLists.txt:27]: thanks. it's terrible to have most RDF tests marked "MIGHT_FAIL", it's slightly less terrible if they are marked "WILL_FAIL_ON_WINDOWS" :)

File: root/dataframe/test_trainCache.C (hunk 1 above)
Comment (eguiraud) [reply root/dataframe/test_trainCache.C:1]: alright, can you please enclose it in an `#ifdef WINDOWS`, if nothing else for documentation purposes?

Comment (eguiraud) [reply root/dataframe/test_trainCache.C:1]: (rationale: we want to keep testing that autoloading works on other platforms)

Comment (bellenot) [reply root/dataframe/test_trainCache.C:1]: Fair enough, will do

File: root/dataframe/CMakeLists.txt (hunk 2 above)
Comment (bellenot) [reply root/dataframe/CMakeLists.txt:27]: According to Axel, it's a known issue in llvm/clang/cling

File: root/dataframe/test_trainCache.C (hunk 1 above)
Comment (bellenot) [reply root/dataframe/test_trainCache.C:2]: Done

**Comment by bellenot** [comment]
> Thanks!

Thanks to you!

**Comment by bellenot** [comment]
Errors are unrelated - merging
